File: app/src/main/java/com/zxcx/zhizhe/ui/my/setting/MessageModeActivity.java
package com.zxcx.zhizhe.ui.my.setting;

import android.os.Bundle;
import android.widget.CheckBox;

import com.zxcx.zhizhe.R;
import com.zxcx.zhizhe.mvpBase.BaseActivity;
import com.zxcx.zhizhe.mvpBase.BaseRxJava;
import com.zxcx.zhizhe.mvpBase.INullGetPostPresenter;
import com.zxcx.zhizhe.retrofit.AppClient;
import com.zxcx.zhizhe.retrofit.BaseBean;
import com.zxcx.zhizhe.retrofit.NullPostSubscriber;
import com.zxcx.zhizhe.utils.SVTSConstants;
import com.zxcx.zhizhe.utils.SharedPreferencesUtil;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnCheckedChanged;
import butterknife.OnClick;

/**
 * Created by anm on 2017/12/13.
 */

public class MessageModeActivity extends BaseActivity implements INullGetPostPresenter<MessageModeBean>{

    @BindView(R.id.cb_message_mode_system)
    CheckBox mCbMessageModeSystem;
    @BindView(R.id.cb_message_mode_dynamic)
    CheckBox mCbMessageModeDynamic;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_message_mode);
        ButterKnife.bind(this);
        initToolBar("消息提示");
        mCbMessageModeSystem.setChecked(SharedPreferencesUtil.getBoolean(SVTSConstants.systemMessageIsOpen,true));
        mCbMessageModeDynamic.setChecked(SharedPreferencesUtil.getBoolean(SVTSConstants.dynamicMessageIsOpen,true));
        getMessageSetting();
    }

    @OnClick(R.id.ll_message_mode_system)
    public void onMLlImageLoadModeWifiClicked() {
        mCbMessageModeSystem.setChecked(!mCbMessageModeSystem.isChecked());
    }

    @OnClick(R.id.ll_message_mode_dynamic)
    public void onMLlImageLoadModeAllClicked() {
        mCbMessageModeDynamic.setChecked(!mCbMessageModeDynamic.isChecked());
    }

    @OnCheckedChanged(R.id.cb_message_mode_system)
    public void onCbSystemChanged() {
        int systemMessageSetting = mCbMessageModeSystem.isChecked()?0:1;
        int dynamicMessageSetting = mCbMessageModeDynamic.isChecked()?0:1;
        setMessageSetting(systemMessageSetting,dynamicMessageSetting);
    }

    @OnCheckedChanged(R.id.cb_message_mode_dynamic)
    public void onCbDynamicChanged() {
        int systemMessageSetting = mCbMessageModeSystem.isChecked()?0:1;
        int dynamicMessageSetting = mCbMessageModeDynamic.isChecked()?0:1;
        setMessageSetting(systemMessageSetting,dynamicMessageSetting);
    }

    public void getMessageSetting(){
        mDisposable = AppClient.getAPIService().getMessageSetting()
                .compose(BaseRxJava.handleResult())
                .compose(BaseRxJava.io_main())
                .subscribeWith(new NullPostSubscriber<MessageModeBean>(this){

                    @Override
                    public void onNext(MessageModeBean bean) {
                        getDataSuccess(bean);
                    }
                });
        addSubscription(mDisposable);
    }

    public void setMessageSetting(int systemMessageSetting,int dynamicMessageSetting){
        mDisposable = AppClient.getAPIService().setMessageSetting(systemMessageSetting,dynamicMessageSetting)
                .compose(BaseRxJava.handlePostResult())
                .compose(BaseRxJava.io_main())
                .subscribeWith(new NullPostSubscriber<BaseBean>(this){

                    @Override
                    public void onNext(BaseBean baseBean) {
                        postSuccess();
                    }
                });
        addSubscription(mDisposable);
    }

    @Override
    public void postSuccess() {
        SharedPreferencesUtil.saveData(SVTSConstants.systemMessageIsOpen,mCbMessageModeSystem.isChecked());
        SharedPreferencesUtil.saveData(SVTSConstants.dynamicMessageIsOpen,mCbMessageModeDynamic.isChecked());
    }

    @Override
    public void postFail(String msg) {
        toastShow(msg);
    }

    @Override
    public void getDataSuccess(MessageModeBean bean) {
        mCbMessageModeSystem.setChecked(bean.isSystemMessageSetting());
        mCbMessageModeDynamic.setChecked(bean.isDynamicMessageSetting());
    }

    @Override
    public void getDataFail(String msg) {
        toastShow(msg);
    }
}

File: app/src/main/java/com/zxcx/zhizhe/ui/my/setting/MessageModeActivity.kt
package com.zxcx.zhizhe.ui.my.setting

import android.os.Bundle
import butterknife.ButterKnife
import com.zxcx.zhizhe.R
import com.zxcx.zhizhe.mvpBase.BaseActivity
import com.zxcx.zhizhe.mvpBase.BaseRxJava
import com.zxcx.zhizhe.mvpBase.INullGetPostPresenter
import com.zxcx.zhizhe.retrofit.AppClient
import com.zxcx.zhizhe.retrofit.BaseBean
import com.zxcx.zhizhe.retrofit.NullPostSubscriber
import com.zxcx.zhizhe.utils.SVTSConstants
import com.zxcx.zhizhe.utils.SharedPreferencesUtil
import kotlinx.android.synthetic.main.activity_message_mode.*

/**
 * Created by anm on 2017/12/13.
 */

class MessageModeActivity : BaseActivity(), INullGetPostPresenter<MessageModeBean> {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_message_mode)
        ButterKnife.bind(this)
        cb_message_mode_system.isChecked = SharedPreferencesUtil.getBoolean(SVTSConstants.systemMessageIsOpen, true)
        cb_message_mode_dynamic.isChecked = SharedPreferencesUtil.getBoolean(SVTSConstants.dynamicMessageIsOpen, true)
        getMessageSetting()
    }

    override fun setListener() {
        iv_common_close.setOnClickListener {
            onBackPressed()
        }

        ll_message_mode_system.setOnClickListener {
            cb_message_mode_system.isChecked = !cb_message_mode_system.isChecked
        }

        cb_message_mode_system.setOnCheckedChangeListener { buttonView, isChecked ->
            val systemMessageSetting = if (cb_message_mode_system.isChecked) 0 else 1
            val dynamicMessageSetting = if (cb_message_mode_dynamic.isChecked) 0 else 1
            setMessageSetting(systemMessageSetting, dynamicMessageSetting)
        }

        ll_message_mode_dynamic.setOnClickListener {
            cb_message_mode_dynamic.isChecked = !cb_message_mode_dynamic.isChecked
        }

        cb_message_mode_dynamic.setOnCheckedChangeListener { buttonView, isChecked ->
            val systemMessageSetting = if (cb_message_mode_system.isChecked) 0 else 1
            val dynamicMessageSetting = if (cb_message_mode_dynamic.isChecked) 0 else 1
            setMessageSetting(systemMessageSetting, dynamicMessageSetting)
        }
    }

    private fun getMessageSetting() {
        mDisposable = AppClient.getAPIService().messageSetting
                .compose(BaseRxJava.handleResult())
                .compose(BaseRxJava.io_main())
                .subscribeWith(object : NullPostSubscriber<MessageModeBean>(this) {

                    override fun onNext(bean: MessageModeBean) {
                        getDataSuccess(bean)
                    }
                })
        addSubscription(mDisposable)
    }

    private fun setMessageSetting(systemMessageSetting: Int, dynamicMessageSetting: Int) {
        mDisposable = AppClient.getAPIService().setMessageSetting(systemMessageSetting, dynamicMessageSetting)
                .compose(BaseRxJava.handlePostResult())
                .compose(BaseRxJava.io_main())
                .subscribeWith(object : NullPostSubscriber<BaseBean<*>>(this) {

                    override fun onNext(baseBean: BaseBean<*>) {
                        postSuccess()
                    }
                })
        addSubscription(mDisposable)
    }

    override fun postSuccess() {
        SharedPreferencesUtil.saveData(SVTSConstants.systemMessageIsOpen, cb_message_mode_system.isChecked)
        SharedPreferencesUtil.saveData(SVTSConstants.dynamicMessageIsOpen, cb_message_mode_dynamic.isChecked)
    }

    override fun postFail(msg: String) {
        toastShow(msg)
    }

    override fun getDataSuccess(bean: MessageModeBean) {
        SharedPreferencesUtil.saveData(SVTSConstants.systemMessageIsOpen, bean.isSystemMessageSetting)
        SharedPreferencesUtil.saveData(SVTSConstants.dynamicMessageIsOpen, bean.isDynamicMessageSetting)
        cb_message_mode_system.isChecked = bean.isSystemMessageSetting
        cb_message_mode_dynamic.isChecked = bean.isDynamicMessageSetting
    }

    override fun getDataFail(msg: String) {
        toastShow(msg)
    }
}
